src/Makefile.inc

```diff
@@ -0,0 +1,340 @@
+# We need to always run this code, as opposed to running it from a prerequisite
+# The HGTEST variable is not used
+HGTEST := $(shell mkdir -p $(ODIR); \
+	[ -f $(ODIR)/hgstamp ] || touch $(ODIR)/hgstamp; \
+	echo $(HGVERSION) | cmp -s $(ODIR)/hgstamp - \
+	|| echo $(HGVERSION) > $(ODIR)/hgstamp)
+############################################
+#END SECTION                               #
+############################################
+
+##################################################
+#This section contains switches used for building#
+##################################################
+ifeq "$(TYPE)" "dev"
+  CFLAGS   := $(CFLAGS) -DBUILDTYPE_DEV
+endif
+ifdef MODULAR
+MODULE_CFLAGS   :=  -DMODULAR=$(MODULAR)
+CFLAGS   +=  -DENABLE_MODULAR=$(MODULAR)
+endif
+
+CFLAGS   := $(CFLAGS) -g -Wall -Wextra -Werror=undef -I$(SDIR) -I$(SDIR)/firmware -D$(PROGMODE) -std=gnu99 \
+            -I$(SDIR)/target/tx/$(FAMILY)/$(TARGET) -I$(SDIR)/gui/$(SCREENSIZE) \
+            -I$(SDIR)/pages/$(SCREENSIZE) \
+            -DHGVERSION="\"${HGVERSION}\""
+CXXFLAGS := $(CXXFLAGS) $(patsubst -std=gnu99,,$(CFLAGS))
+EXEEXT   ?= elf
+##################################################
+#END SECTION                                     #
+##################################################
+
+############################################
+#this section defines final build files    #
+############################################
+FILESYSTEM = $(subst emu_,,$(TARGET))
+MODELDIR = filesystem/$(FILESYSTEM)/models
+
+LAST_MODEL := $(MODELDIR)/model$(NUM_MODELS).ini
+
+ALL += $(TARGET).fs_wrapper
+
+############################################
+#END SECTION                               #
+############################################
+
+############################################
+#this section includes build targets       #
+############################################
+all : $(ALL)
+
+####################################
+# recompile if the Makefile changes#
+####################################
+#$(OBJS) $(PROTO_OBJS) ($PROTO_EXTRA_OBJS): Makefile target/tx/$(FAMILY)/$(TARGET)/Makefile.inc $(EXTRA_MAKEFILES)
+
+##################################################################################
+# The following enables quiet output unless you use VERBOSE=1                    #
+# Note that this must be after the 1st rule so that it doesn't execute by default#
+##################################################################################
+$(VERBOSE).SILENT:
+
+.PHONY: clean language lint
+
+clean:
+	rm -f $(TARGET).$(EXEEXT) $(TARGET).exe $(TARGET).bin $(TARGET).dfu $(TARGET).list \
+		$(TARGET).map $(ODIR)/*.o $(ODIR)/*.o_ $(ODIR)/*.P  $(ODIR)/*.bin \
+		filesystem/$(FILESYSYTEM) 2> /dev/null || true
+
+language:
+	../utils/extract_strings.pl -po > $(SDIR)/fs/language/deviation.po
+
+
+zip: $(TARGET).zip
+
+%.zip: $(ALL) $(TARGET).dfu $(PROTO_MODULES)
+	#This is not an emulator build (emulator is hanled in target/common/emu/Makefile.inc)
```

Comment:
knowledge of anything dfu related is target based and shouldn't be in the main file....but tis the general zip logic common to all tx?  it is different for emus....it is at least somewhat different for radiolnk and opentx. Not sure if we can break this up or move it without making it either duplicated or hard to read.